### ISSUE TITLE
IP-Addresses in DNS subjectAltName extension makes verify_certificate_hostname() error out

### ISSUE THREAD

**Issue by schlenk** [pyca/service-identity#35]
I tried to use service-identity to do a configuration check for a server.

Basically checking if the configured names are available in the provided certificate.

````python
    cert = x509.load_pem_x509_certificate(stream.read(), default_backend())
    for hostname in hostnames:
         svcid.cryptography.verify_certificate_hostname(cert, hostname)
````

This does not work, if the certificate contains dual entries for an IP both in IPAddress and in a DNS field in the SAN, because it throws an "CertificateError: Invalid DNS pattern". This might be not allowed by the CAB rules (as mentioned in #12 ), but is not uncommon for inhouse CAs trying to support legacy systems.

So even if the correct hostname is included in the certificate, i cannot check it, due to the unrelated issue of IPs in the SAN throwing an Exception.

It would be nice, if the API would provide a way to collect all issues found instead of raising an exception on the first issue. This would make the API more useful for server side configuration checkers instead of client side checking only.

**Comment by agormley** [comment]
Ran into this same issue and worked around with the following change by removing the check for IP in DNSPattern:
`--- before/service_identity-17.0.0/src/service_identity/_common.py    2017-05-23 18:22:50.000000000 +0000
+++ service_identity-17.0.0/src/service_identity/_common.py    2019-08-30 20:50:27.056000000 +0000
@@ -151,7 +151,7 @@
        pattern = pattern.strip()
-        if pattern == b”" or _is_ip_address(pattern) or b”\0" in pattern:
+        if pattern == b”" or b”\0" in pattern:
            raise CertificateError(
                “Invalid DNS pattern {0!r}.“.format(pattern)
            )
`

**Comment by hynek** [comment]
I think I would accept a PR that would permit this via a flag that is False by default.

**Comment by schlenk** [comment]
Fixed with 18.1.0
